Use our internal gswin32c.exe (9.21) for ghostscript processing

StephenMcConnel opened the pull request:
In case we have to stop shipping it, enforce at least version 9.18
before using it in processing.  We know 9.18 works (on Linux) and 9.16
fails (on Windows).

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/bloombooks/bloomdesktop/2034)
<!-- Reviewable:end -->

JohnThomson commented:
Basically good, Couple of minor points, also please check that we are actually satisfying the requirements for aggregation (as unobtrusively as possible). We talked about having a copy of the AGPL and a ReadMe.htm with links to the full installer and the exact-version source in the ghostscript folder. Also, please make sure Bloom's About box mentions ghostscript and says whatever the license requires. (Maybe it already does.)

---

Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[src/BloomExe/Publish/PDF/ProcessPdfWithGhostscript.cs, line 113 at r1](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/2034#-L0PxanS28OawJY_Y1N9:-L0PxanS28OawJY_Y1NA:b-l227a2) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/ee250277ed6f8c8dc09b305cba5e56e83ad6dec9/src/BloomExe/Publish/PDF/ProcessPdfWithGhostscript.cs#L113)):*
> ```Smalltalk
> 			}
> 			// TODO: if we decide to distribute GS with Bloom, enhance this to look first wherever we
> 			// install GS in the user area.
> ```

It appears you have just done this ToDo, so please remove it.

---

*[src/BloomExe/Publish/PDF/ProcessPdfWithGhostscript.cs, line 133 at r1](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/2034#-L0PxmR2j0rZ_Wf0cfAy:-L0PxmR349X810EH8XnK:bt1k9ey) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/ee250277ed6f8c8dc09b305cba5e56e83ad6dec9/src/BloomExe/Publish/PDF/ProcessPdfWithGhostscript.cs#L133)):*
> ```Smalltalk
> 				// gs9.18 works on Linux.  gs9.16 fails on Windows.  See BL-5295.
> 				var version = Path.GetFileName(versionDir);
> 				if (string.Compare(version, "gs9.18", StringComparison.Ordinal) < 0)
> ```

I think this will reject gs10.X if there ever is one, which is probably not what we want...

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/2034#-:-L0PyAD8CTVzSc66zVuB:b-rj5lgz)*
<!-- Sent from Reviewable.io -->

StephenMcConnel commented:
I've added an HTML Readme file plus a plain text copy of the AGPL to DistFiles/ghostscript and made the suggested changes to the code.

---

Review status: 0 of 4 files reviewed at latest revision, 2 unresolved discussions.

---

*[src/BloomExe/Publish/PDF/ProcessPdfWithGhostscript.cs, line 113 at r1](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/2034#-L0PxanS28OawJY_Y1N9:-L0QuwqCkq8_Msw-qtkd:b-896fix) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/ee250277ed6f8c8dc09b305cba5e56e83ad6dec9/src/BloomExe/Publish/PDF/ProcessPdfWithGhostscript.cs#L113)):*
<details><summary><i>Previously, JohnThomson (John Thomson) wrote…</i></summary><blockquote>

It appears you have just done this ToDo, so please remove it.
</blockquote></details>

Done.

---

*[src/BloomExe/Publish/PDF/ProcessPdfWithGhostscript.cs, line 133 at r1](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/2034#-L0PxmR2j0rZ_Wf0cfAy:-L0QuRPJQwhDIXGv246r:ba6oadh) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/ee250277ed6f8c8dc09b305cba5e56e83ad6dec9/src/BloomExe/Publish/PDF/ProcessPdfWithGhostscript.cs#L133)):*
<details><summary><i>Previously, JohnThomson (John Thomson) wrote…</i></summary><blockquote>

I think this will reject gs10.X if there ever is one, which is probably not what we want...
</blockquote></details>

Fixed code to compare float values.  Upped minimum version to 9.21.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/2034#-:-L0R7nkPo1kSiKOWlU6K:b-l2f798)*
<!-- Sent from Reviewable.io -->

JohnThomson commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 4 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/2034#-:-L0RCNIt2b7Lk4rRXVT2:bnfp4nl)*
<!-- Sent from Reviewable.io -->

JohnThomson commented:
Reviewed 4 of 4 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/2034)*
<!-- Sent from Reviewable.io -->
